### ISSUE TITLE
NamedNumpyArray doesn't work with np.stack (or indexing with None in general)

### ISSUE THREAD

**Issue by jhostetler** [deepmind/pysc2#273]
[pysc2 2.0.2, numpy 1.14.1]

Indexing a `numpy.ndarray` with `None` will create a new axis in the array. This mechanism is used in some functions like `np.stack`. `NamedNumpyArray` doesn't accept `None` as an index:

```
import numpy as np
from pysc2.lib.named_array import NamedNumpyArray
  
a = NamedNumpyArray( [1, 2, 3], ["x", "y", "z"] )
b = NamedNumpyArray( [4, 5, 6], ["x", "y", "z"] )

# This works:
# a = np.expand_dims( a, axis=0 )
# b = np.expand_dims( b, axis=0 )
# c = np.concatenate( [a, b], axis=0 )

# This fails:
c = np.stack( [a, b], axis=0 )
```

This will produce the following error:
```
File "<path>\site-packages\numpy\core\shape_base.py", line 359, in <listcomp>
    expanded_arrays = [arr[sl] for arr in arrays]
  File "<path>\site-packages\pysc2\lib\named_array.py", line 182, in __getitem__
    raise TypeError("Unknown index: %s; %s" % (type(index), index))
TypeError: Unknown index: <class 'NoneType'>; None
```

**Comment by ethan052** [comment]
I guess that NamedNumpyArray is used for providing multiple ways to access elements, and providing some operations, such like masking, as np.ndarray .
As your example, you can use a[1], a["y"], or a.y to retrieve the same memory space and its value is 2.

Although you can use np.expand_dims and np.concatenate to simulate the effect of np.stack,
the string indexed and field like style accessing ways would be meaningless.

If you want to use full functionality of np.ndarray, you may use its constructor to convert type from NamedNumpyArray to np.ndarray.

**Comment by tewalds** [comment]
This should be fixed in e994d8b

**Comment by verystrongjoe** [comment]
Any update?
